src/PiWeb.Volume.Convert/ConvertVolume.cs

```diff
@@ -76,26 +69,61 @@ public static Volume FromVgi(
 			}
 		}
 
+		/// <summary>
+		/// Creates an uncompressed volume from vgi header stream and a data stream that contains uint8 or uint16 values
+		/// that represent grayscale voxels. The voxels must be ordered in z, y, x direction.
+		/// (slice by slice, row by row).
+		/// </summary>
+		public static Volume FromGomVolume(
+			VolumeMetadata metadata,
+			Stream dataStream,
+			Gom.DataType dataType,
+			bool extraPolate,
+			double minValue,
+			double maxValue,
+			bool streamed,
+			IProgress<double>? progress = null,
+			ILogger? logger = null )
+		{
+			var sw = Stopwatch.StartNew();
+
+			try
+			{
+				var voxelStream = dataType switch
```

Comment:
might be easier to use a tuple here: 

```cs
(dataType, extraPolate) switch
{
  ( Gom.DataType.Int16, false ) => ...
}
```

Reply:
I don't think this improves the readability